core/src/main/scala/org/apache/spark/scheduler/TaskDescription.scala

```diff
@@ -52,7 +55,36 @@ private[spark] class TaskDescription(
     val addedFiles: Map[String, Long],
     val addedJars: Map[String, Long],
     val properties: Properties,
-    val serializedTask: ByteBuffer) {
+    private var serializedTask_ : ByteBuffer,
+    private var task_ : Task[_] = null) extends  Logging {
```

Comment:
yeah, that leads to ~100 compile errors in test code.  its all small changes but still, not sure if there is much value.  I don't understand the comment on LaunchTask -- your commit does not change it, right?

in any case, if you feel like we should do that refactor, I am not opposed.